insights/parsers/systemd_analyze.py

```diff
@@ -0,0 +1,74 @@
+"""
+SystemdAnalyzeBlame - command ``systemd-analyze blame``
+=======================================================
+
+This module parses the output of command ``systemd-analyze blame``.
+"""
+from insights.specs import Specs
+from insights import CommandParser, parser
+from insights.parsers import keyword_search
+
+
+@parser(Specs.systemd_analyze_blame)
+class SystemdAnalyzeBlame(CommandParser):
+    """Parse the output of ``systemd-analyze blame``.
+
+    Typical output::
+
+        33.080s cloud-init-local.service
+        32.423s unbound-anchor.service
+         2.773s kdump.service
+         1.699s dnf-makecache.service
+         1.304s cloud-init.service
+         1.073s initrd-switch-root.service
+          939ms cloud-config.service
+          872ms tuned.service
+          770ms cloud-final.service
+
+    Examples:
+
+        >>> 'cloud-init-local.service' in output
+        True
+        >>> output.search('cloud-init-local.service') == {'time': 33.08, 'unit_name': 'cloud-init-local.service'}
+        True
+        >>> output.search('cloud-init.service').get('time')
+        1.304
+        >>> output.search('foobar.service')
+        >>> 'foobar.service' in output
+        False
+
+    Attributes:
+        data (list): List of key-value pair for each Systemd Unit.
+                     The time to initialize is converted to Seconds.
+                     Ex: ``[{'time': 0.08, 'unit_name': 'auditd.service'}]``
+        units (list): Unit names from the output.
+    """
+    def parse_content(self, content):
+        self.data = []
+        self.units = []
+        for c in content:
+            time, service = c.split()
+            if time.endswith('ms'):
+                _time = round(float(time.strip('ms')) / 1000, 5)
+                self.data.append({'time': _time, 'unit_name': service})
+            else:
+                _time = round(float(time.strip('ms')), 5)
+                self.data.append({'time': _time, 'unit_name': service})
+
+        self.units = [i['unit_name'] for i in self.data]
+
+    def __contains__(self, kw):
+        return kw in self.units
+
+    def search(self, unit_name):
+        """Search unit name.
+
+        Example:
+
+            >>> output.search('cloud-init-local.service') == {'time': 33.08, 'unit_name': 'cloud-init-local.service'}
+            True
+            >>> output.search('cloud-init.service').get('time')
+            1.304
+        """
+        if unit_name in self.units:
+            return keyword_search(self.data, unit_name=unit_name)[-1]
```

Comment:
@psachin this is a little confusing.  You have to match the full unit_name to get to this keyword search.  In a rule you could instead have used `"unit_name" in output.data`, and accessed the data with `output.data["unit_name"]`.  The `keyword_search` doesn't provide any extra functionality here since the full `unit_name` has to match and its the actual key in the dictionary which has to be unique.  Maybe instead you meant to implement a search for substrings in the `unit_name`?  Also search should return an empty list instead of None if nothing is found to be consistent.  This should be indicated in the method docstring.